python/sdk/prelude_sdk/controllers/iam_controller.py

```diff
@@ -59,205 +48,191 @@ def update_account(self, mode: Mode = None, company: str = None, slug: str = Non
+        res = self.get(
+            f"{self.account.hq}/iam/user/account",
+            headers=self.account.headers,
+            timeout=10,
         )
-        if res.status_code == 200:
-            cfg = self.account.read_keychain_config()
-            res_json = res.json()
-            cfg[self.account.profile]["account"] = res_json["account"]
-            cfg[self.account.profile]["token"] = res_json["token"]
-            self.account.write_keychain_config(cfg)
-            return res_json
-        raise Exception(res.text)
+        return res.json()
 
     @verify_credentials
     def audit_logs(self, days: int = 7, limit: int = 1000):
         """Get audit logs from the last X days"""
         params = dict(days=days, limit=limit)
-        res = self._session.get(
+        res = self.get(
             f"{self.account.hq}/iam/audit",
             headers=self.account.headers,
             params=params,
             timeout=30,
         )
-        if res.status_code == 200:
-            return res.json()
-        raise Exception(res.text)
+        return res.json()
 
     @verify_credentials
-    def subscribe(self, event: AuditEvent):
-        """Subscribe to email notifications for an event"""
-        res = self._session.post(
-            f"{self.account.hq}/iam/subscribe/{event.name}",
-            headers=self.account.headers,
-            timeout=10,
-        )
-        if res.status_code == 200:
-            return res.json()
-        raise Exception(res.text)
+    def get_oidc_name(self, slug: str):
+        """Get OIDC provider name from organization slug"""
+        params = dict(slug=slug)
 
-    @verify_credentials
-    def unsubscribe(self, event: AuditEvent):
-        """Unsubscribe to email notifications for an event"""
-        res = self._session.delete(
-            f"{self.account.hq}/iam/subscribe/{event.name}",
+        res = self.get(
+            f"{self.account.hq}/iam/account/oidc",
             headers=self.account.headers,
+            params=params,
             timeout=10,
         )
-        if res.status_code == 200:
-            return res.json()
-        raise Exception(res.text)
+        return res.json()
 
     @verify_credentials
-    def accept_terms(self, name, version):
-        """Accept terms and conditions"""
-        res = self._session.post(
-            f"{self.account.hq}/iam/terms",
+    def sign_up(self, company, email, name):
+        """Create a new user and account"""
+        body = dict(company=company, email=email, name=name)
+
+        res = self.post(
+            f"{self.account.hq}/iam/new_user_and_account",
             headers=self.account.headers,
-            json=dict(name=name, version=version),
-            timeout=10,
+            json=body,
+            timeout=20,
+        )
+        data = res.json()
+        self.account.keychain.configure_keychain(
+            data["account_id"], data["user_id"], self.account.hq, company
```

Comment:
can the profile name in the keychain be a separate input param that's optional (so that 1. i dont accidentally overwrite a profile for the same company and 2. it doesnt try to much about with a keychain if its not set)